src/bijectors/stacked.jl

```diff
@@ -80,12 +80,15 @@ function (sb::Stacked{<:AbstractArray, N})(x::AbstractVector{<:Real}) where {N}
     return y
 end
 
-# HACK: `reshape` to get around the fact that `hcat` isn't type-stable
-(sb::Stacked)(x::AbstractMatrix{<:Real}) = reshape(foldl(hcat, [sb(x[:, i]) for i = 1:size(x, 2)]), size(x)...)
+@views function (sb::Stacked)(x::AbstractMatrix{<:Real})
+    init = reshape(sb(x[:, 1]), :, 1)
+    return mapreduce(i -> sb(x[:, i]), hcat, 2:size(x, 2); init = init)
```

Comment:
No need to define `init` separately. This is equivalent to using `1:size(x, 2)` and dropping the `init` kwarg.

Reply:
Hmm actually I double checked and your implementation here is type stable but the compiler is struggling to infer my suggestion.

Reply:
Yes, this change is only for fixing the type inference problem.